app/controllers/tweets_controller.rb

```diff
@@ -0,0 +1,57 @@
+class TweetsController < ApplicationController
+#==============================tweets================================ 
+  get '/tweets' do 
+    redirect '/login' if !logged_in?
+    
+    erb :"tweets/tweets"
+  end
+#===============================new================================== 
+  get '/tweets/new' do 
+    redirect '/login' if !logged_in?
+    
+    erb :'tweets/create_tweet'
+  end
+#-------------------------------------------------------------------- 
+  post '/tweets' do 
+    redirect '/tweets/new' if empty?(params)
+    
+    Tweet.create(content: params[:content], user_id: current_user.id)
+    redirect '/tweets'
+  end
+#==============================show================================== 
+  get '/tweets/:id' do 
+    redirect '/login' if !logged_in?
+    
+    @tweet = tweet(params[:id])
+    
+    erb :'/tweets/show_tweet'
+  end
+#==============================edit================================== 
+  get '/tweets/:id/edit' do 
+    redirect '/login' if !logged_in? 
+    
+    @tweet = tweet(params[:id])
+    
+    redirect '/tweets' if @tweet.user_id != current_user.id  
+    erb :'/tweets/edit_tweet'
+  end
+#=============================update================================= 
+  patch '/tweets/:id' do 
+    @tweet = tweet(params[:id])
+    redirect "/tweets/#{@tweet.id}/edit" if params[:content].empty?
+    
+    @tweet.update(content: params[:content])
+    redirect "/tweets/#{@tweet.id}"
+  end
+#=============================delete================================= 
+  delete '/tweets/:id/delete' do 
+    @tweet = tweet(params[:id])
+    
+    @tweet.destroy! if @tweet.user_id == current_user.id
```

Comment:
Prefer to find tweets that are already only associated to the `current_user` than to manually check the `user_id` foreign key of the tweet against the `current_user.id`

```ruby
tweet = current_user.tweets.find_by(id: params[:id])
if tweet && tweet.destroy
  redirect "/tweets"
else
  redirect "/tweets/#{tweet.id}"
end
```

We're using the association collection method provided by [`has_many`](http://api.rubyonrails.org/classes/ActiveRecord/Associations/ClassMethods.html#method-i-has_many) to query the user's tweets. 

Note the usage of [`find_by`](http://api.rubyonrails.org/classes/ActiveRecord/FinderMethods.html#method-i-find-by) vs [`find`](http://api.rubyonrails.org/classes/ActiveRecord/FinderMethods.html#method-i-find). Had we used `find` here, ActiveRecord would raise an error in the event that we could not find a tweet associated with this user. `find_by` will allow this action to fail gracefully, simply returning nil if the user is trying to delete a tweet that they do not have access to.

Reply:
that's an awesome tip especially since I default to using find() when only need the id as it's prettier. Will definitely keep this in mind :).